### DISCUSSION TITLE
Route Query Lost After Page Refresh

### DISCUSSION THREAD

**Discussion by darrylmorley** [vercel/next.js#17959] [Help]
Hi all,

I'm using a route query to pass a product ID through to my product page, allowing me to pull all of the related info in getInitialProps.

This works well until the page is refreshed, when the page is refreshed the query id is lost and the product info can no longer be pulled.

Is there a way to persist the query after refresh?

Or is there another approach that I should be using?

Thanks

**Comment by hasanzahran** [comment]
I have the same issue!

**Reply by darrylmorley** [reply]
Give me a shout if you resolve it, I've not got around to it yet 👍

**Reply by jibaromar** [reply]
Hello, did you manage to resolve the issue? I've got the same question.

**Reply by javujavichi** [reply]
Hi, I've got the same question, any updates?

**Reply by darrylmorley** [reply]
I don't know the answer to this unfortunately, I've been using getStaticProps since I opened this. Maybe getStaticProps/Paths would work for you? 

You could always use localStorage or cookies if all else fails.

**Reply by leerob** [reply]
I would recommend using `getServerSideProps` for this if you need fresh data on every request. If you can statically generate, then you `getStaticProps`/`getStaticPaths`.
